### DISCUSSION TITLE
Probing is not updating coordinates

### DISCUSSION THREAD

**Discussion by R-Ronald** [terjeio/ioSender#265] [Q&A]
I've got a PrintNC. Several folks over on the PrintNC discord have been attempting to help resolve this problem w/o success.

I turn the machine on, do a home, jog to my work piece and execute a probe. 

- If I do a Z only probe, it seems to complete (does initial probing and a latch probe before returning to the original starting position)......but nothing is updated
- If I do an external corner probe, it does a little dance and probes both edges. Nothing is updated. And, even if I check-mark the Probe Z box, nothing is updated. The probe finishes off to one side of the part, not at the corner (I see in other videos that it finishes at the corner).

I've done this without establishing a TLO reference and after establishing a TLO reference. No change.

I encountered this problem when running IOSender 2.0.40. Last night I updated to 2.0.41, with no improvement.

One other note. I don't seem to be able to change the DRO values in any way. The zero (and zero all) buttons don't do anything. And if I enter a new value, the box reverts back to the original value.

The machine homes/jogs, sets offsets, and executes .nc programs w/o any issues.

**Comment by terjeio** [comment]
This must be a controller issue, I have a STM32F756 and a RP2040 board here where I can set the DRO values and zero just fine. I am not able to test probing since I do not have access to a physical CNC (until late January).

Have you tested with the latest official grblHAL build?

**Comment by R-Ronald** [comment]
I held off a bit to test a Jog2K that was on the way before messing with the grblHAL firmware. I received it yesterday and thus proceeded with this issue today.

This afternoon, I flashed the onboard Teensy with the latest grblHAL build (the Oct 2, 2022 binary from https://github.com/Expatria-Technologies/iMXRT1062/releases). I tested and did not see any changes in behavior. The values shown in the ioSender screen don't change when any of the axes zero buttons or the "zero all" button is clicked.

However, since I now have a Jog2K connected, I have a bit more information. When I click one of the DRO zero buttons on the windows screen, I do see a console message from ioSender. And the jogger's DRO values are updated properly. So, the controller is receiving the update and passing it on to the Jog2K. It's just the visible DRO on the ioSender GRBL page that refuses to change.

I didn't retest the edge finding (after re-flashing the controller) yet.

**Reply by terjeio** [reply]
Could it be missing bits in the $10 setting? See [this issue](https://github.com/grblHAL/RP2040/issues/46).

**Reply by R-Ronald** [reply]
You've solved my problem!

I had "19" in parameter 10. No idea why, as I'm pretty certain I never changed/set any of those bits.

Putting "511" into the parameter not only gave me access to allow changing the DRO values, it also fixed my probing issues. I suspect that once the code realized it couldn't update values after probing the edges, it gave up before trying to do the Z probe. In any event, it works now and I'm very happy.

Thanks a bunch!
